lib/puma/thread_pool.rb

```diff
@@ -208,6 +213,24 @@ def wait_until_not_full
       end
     end
 
+    def wait_for_less_busy_worker
+      # Ruby MRI does GVL, this can result
+      # in processing contention when multiple threads
+      # (requests) are running concurrently
+      return unless Puma.mri?
+
+      with_mutex do
+        return if @shutdown
+
+        # do not delay, if we are not busy
+        return unless busy_threads > 0
+
+        # this will be signaled once a request finishes,
+        # which can happen earlier than delay
+        @not_full.wait @mutex, BUSY_WORKER_DELAY
```

Comment:
Are you running this particular variation in prod? It looks like you decided the ticks concept/more-loaded-workers-wait-longer idea was not necessary.

Reply:
It would be cool if we could set/get watcher priority in libev through nio4r, then we could do this w/o any latency at all. But that's `TODO`.

Reply:
Not this particular one, once we agree that this is OK. I will push this change to `gitlab.com` and measure it there.